packages/compass-components/src/components/databases-navigation-tree.tsx

```diff
@@ -0,0 +1,712 @@
+        )}
+      </div>
+    </li>
+  );
+};
+
+const fadeInAnimation = keyframes({
+  '0%': {
+    opacity: 0,
+  },
+  '100%': {
+    opacity: 1,
+  },
+});
+
+const fadeIn = css({
+  animation: `${fadeInAnimation} .16s ease-out`,
+});
+
+const CollectionsList: React.FunctionComponent<{
+  collections: Collection[];
+  onNamespaceAction(namespace: string, action: Actions): void;
+  activeNamespace?: string;
+  isReadOnly: boolean;
+}> = ({ collections, onNamespaceAction, activeNamespace, isReadOnly }) => {
+  return (
+    <ul role="group" className={ulReset}>
+      {collections.map((coll, index) => (
+        <CollectionItem
+          key={coll._id}
+          id={coll._id}
+          name={coll.name}
+          type={coll.type}
+          posInSet={index + 1}
+          setSize={collections.length}
+          onNamespaceAction={onNamespaceAction}
+          isActive={coll._id === activeNamespace}
+          isReadOnly={isReadOnly}
+        />
+      ))}
+    </ul>
+  );
+};
+
+const placeholderContainer = css({
+  position: 'absolute',
+  pointerEvents: 'none',
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  opacity: 1,
+});
+
+const placeholderContainerHidden = css({
+  opacity: 0,
+  transition: 'opacity .16s ease-out',
+});
+
+const placeholderItem = css({
+  display: 'flex',
+  alignItems: 'center',
+  height: COLLECTION_ROW_HEIGHT,
+  paddingLeft: spacing[5],
+});
+
+const placeholderItemContent = css({
+  display: 'block',
+  height: spacing[3],
+  backgroundColor: uiColors.gray.dark1,
+  borderRadius: 3,
+});
+
+function getBoundRandom(min: number, max: number) {
+  return Math.random() * (max - min) + min;
+}
+
+const PlaceholderItem: React.FunctionComponent = () => {
+  const width = useMemo(() => {
+    return `${getBoundRandom(30, 80)}%`;
+  }, []);
+
+  return (
+    <li className={placeholderItem}>
+      <span style={{ width }} className={placeholderItemContent}></span>
+    </li>
+  );
+};
+
+const CollectionsListPlaceholder: React.FunctionComponent<{
+  length: number;
+  isVisible: boolean;
+}> = ({ length, isVisible }) => {
+  const items = useMemo(() => {
+    return Array.from({ length }, (_, index) => (
+      <PlaceholderItem key={index}></PlaceholderItem>
+    ));
+  }, [length]);
+
+  return (
+    <ul
+      role="presentation"
+      data-testid="placeholder"
+      className={cx(
+        ulReset,
+        placeholderContainer,
+        !isVisible && placeholderContainerHidden
+      )}
+    >
+      {items}
+    </ul>
+  );
+};
+
+const actionsContainer = css({
+  flex: 'none',
+  marginLeft: 'auto',
+  alignItems: 'center',
+  display: 'flex',
+});
+
+const databaseItem = css({
+  height: DATABASE_ROW_HEIGHT,
+  paddingLeft: spacing[1],
+  paddingRight: spacing[1],
+  display: 'flex',
+  alignItems: 'center',
+  backgroundColor: backgroundColor,
+});
+
+const activeBackground = css({
+  backgroundColor: backgroundColorActive,
+});
+
+const databaseItemContainer = css({
+  contentVisibility: 'auto',
```

Comment:
As we are now on a pretty modern version of Chromium I though that we might try using [content-visibility](https://developer.mozilla.org/en-US/docs/Web/CSS/content-visibility) instead of the react-virtualized. I didn't notice any significant performance degradation, even the opposite as we are doing much less relayouts now and don't need to watch for the container size changes, but please try it out locally and tell me if you see any weird behaviors

(I also probably need to add this to the collection list explicitly as the user might have small amount of databases with enormous amounts of collections)